articles/openshift/oshift-ref-dns-zone-forwarding.md

```diff
@@ -0,0 +1,68 @@
+---
+
+# DNS Zone Forwarding in OpenShift Clusters
+
+## Overview
+
+This article documents the ability to forward DNS zones within OpenShift deployments. Forwarding zones can assist in resolving services on private networks if you have existing private DNS resolvers configured within your wider UKCloud environment. This removes the need to individually add domains to a node's local host file and can aid consumption of private services from within OpenShift.
+
+### Intended audience
+
+This article is best aimed at those with OpenShift deployments with connectivity to a VRF as other networks commonly have open resolvers available.
+
+## OpenShift Cluster DNS Architecture
+
+We deploy HAProxy load balancer pairs as part of an OpenShift deployment; primarily to provide redundancy for inbound traffic. As a minimum, all clusters will have: 
+
+- control plane load balancers forwarding to routes:
+  - OpenShift API/Web Console
+  - Docker Registry Console
+  - Grafana
+  - Hawkular Metrics
+  - Kibana
+  - Prometheus
+
+- data plane load balancers forwarding to routes exposed on the default OpenShift routers
+
+Deployments with connectivity to government community networks have an additional data plane load balancer pair (forwarding to routes exposed on the secondary OpenShift routers) which only have inbound connectivity from the secondary network.
+
+On the control plane load balancers, we configure `BIND` (also referred to as `named`) to resolve the local zone within the cluster (performing internal hostname resolution) and all other zones are forwarded outbound to an open internet resolver. All internet OpenShift nodes reference the control plane load balancers as resolvers.
+
+Where a deployment has connectivity with government community networks, we forward the local zone and a small number of internet zones (required for provisioning and ongoing maintenance) to the control plane BIND service, with all remaining zones being forwarded to open resolvers on the secondary network. All secondary network nodes reference the secondary network data plane load balancers as resolvers.
+
+## Identifying Forward Zones
+
+A zone is a domain (for example domain.com) and configuring a forward zone will forward any DNS queries received for that domain and all sub-domains to the configured IP address. This allows different domains to be resolved by separate DNS servers at a granular level if required.
+
+In OpenShift deployments with connectivity to a VRF, you may want to resolve a pre-existing internal zone using a private resolver, instead of forwarding these requests to an external service (and therefore routing traffic over an external network) which gives you the benefit of being able to locate private services that may otherwise be restricted or inaccessible from these networks. Such examples of services that could be consumed from OpenShift include: container image registry, binary repository, version control system.
```

Comment:
I think we should move from VRF to private connectivity or something similar. Then we'll save effort once IRC comes in.